openedx/core/djangoapps/programs/tasks.py

```diff
@@ -706,64 +723,121 @@ def _retry_with_custom_exception(username, course_key, reason, countdown):
 @set_code_owner_attribute
 def update_certificate_visible_date_on_course_update(self, course_key):
     """
-    This task is designed to be called whenever a course is updated with
-    certificate_available_date so that visible_date is updated on credential
-    service as well.
+    This task is designed to be called whenever a course-run's `certificate_available_date` is updated.
```

Comment:
Changes from here on may be a little harder to read in an over/under diff. 

I was not excited that the `update_certificate_visible_date_on_course_update` task had _two_ responsibilities:
* Queue a Celery task to update the `certificate_available_date` for the course certificate configuration in the Credentials IDA, and,
* Queue a number of Celery subtasks to update the `visible_date` attribute for each credential issued to learners in the specified course-run.

With the addiitional logic needed to update course certificate configurations in self-paced course-runs, I thought it was a good time to split this task into two distinct tasks; one dedicated to course certificate configuration updates, and the other for visible_date attribute updates.